src/NSubstitute/Core/CallSpecification.cs

```diff
@@ -45,12 +45,13 @@ private static bool AreComparable(MethodInfo a, MethodInfo b)
             return true;
         }
 
-        if (a.IsGenericMethod && b.IsGenericMethod)
+        if (!a.GetType().IsAssignableFrom(b.GetType()) && !b.GetType().IsAssignableFrom(a.GetType()))
```

Comment:
GetType is an expensive call AFAIK. Maybe create variables for those? (Saves two times the lookup, and is also easier to read/debug

Reply:
Improved, and this spurred me to notice I wasn't actually checking what I intended to